### PR TITLE
#279 Implemented a minTime in TimeService

### PR DESCRIPTION
**Pull request by TijnBertens**
#### What have I done
Implemented a minimum time in the TimeService, which is set to be the time of the first scheduled moment of all sheets. The slider now starts at this minimum time.

#### Testing
Run test cases, inspect animation controls in the application.

#### Addresses SRD section(s)
TimeService
TimeParser?

### CODE REVIEW

File: src/app/backend/file-parser/timing-parser.service.ts [hunk 1]
```diff
@@ -128,12 +138,19 @@ export class TimingParserService {
...
+        }
+
         // Check if a valid maximum timestamp has been found. Zero is also considered to be a
         // valid maximum as a single frame could contain valuable information still
         if (maxTime === undefined || maxTime < 0) {
             throw new ParserError('No valid maximum timestamp found');
         }
 
+        this.timeService.setMinTime(minTime);
+
```
Comment (Dacaspex) [inline src/app/backend/file-parser/timing-parser.service.ts]: Redundant newline

File: src/app/workspace/visualisation-control/slider/slider.component.ts [hunk 2]
```diff
@@ -42,7 +42,9 @@ export class SliderComponent implements OnInit {
      */
     updateTimeFromSlider(time) {
         const placeSlider = time / this.maxSliderValue;
-        this.time.setCurrentTime(placeSlider * this.time.getMaxTime());
+        this.time.setCurrentTime(placeSlider *
+            (this.time.getMaxTime() - this.time.getMinTime()) +
+            this.time.getMinTime());
     }
```
Comment (Dacaspex) [inline src/app/workspace/visualisation-control/slider/slider.component.ts]: You can improve the formatting here
```
this.time.setCurrentTime(
    placeSlider * (this.time.getMaxTime() - this.time.getMinTime()) + this.time.getMinTime()
);
```

File: src/app/backend/time/time.service.ts [hunk 3]
```diff
@@ -174,14 +199,14 @@ export class TimeService {
...
+        return 100 * (this.getCurrentTime() - this.getMinTime()) / (this.getMaxTime() - this.getMinTime());
     }
 
     /**
      * Returns whether the animation can be played back further.
      */
     playbackValid(): boolean {
-       return !((this.getPercentageDone() >= 100 && this.getPlaybackSpeed() > 0) ||
-           (this.getPercentageDone() <= 0 && this.getPlaybackSpeed() < 0));
+        return !((this.getPercentageDone() >= 100 && this.getPlaybackSpeed() > 0) ||
```
Comment (Dacaspex) [inline src/app/backend/time/time.service.ts]: Complex expression, perhaps add a comment? 

I know you did not make this code, but it's nice to leave things clean

Comment (Extremelyd1) [reply src/app/backend/time/time.service.ts]: Added comment.

File: src/app/workspace/visualisation-control/slider/slider.component.ts (hunk 2 above)
Comment (Extremelyd1) [reply src/app/workspace/visualisation-control/slider/slider.component.ts]: Improved formatting.

File: src/app/backend/file-parser/timing-parser.service.ts (hunk 1 above)
Comment (Extremelyd1) [reply src/app/backend/file-parser/timing-parser.service.ts]: Removed newline.

**Pull request merged**